packages/react/src/ReactElementValidator.js

```diff
@@ -221,7 +221,14 @@ function validatePropTypes(element) {
   }
   var name = componentClass.displayName || componentClass.name;
   var propTypes = componentClass.propTypes;
-
+  if (componentClass.PropTypes !== undefined) {
```

Comment:
We should deduplicate this warning so it doesn't warn more than once. See how we do with most other warnings.

Reply:
I have added the boolean flag after observing other places where `warning()` was used. Also tested locally